### PR TITLE
Issue #4675: increase coverage of pitest-checks-misc to 93%

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4675

[pitest-checks-misc before changes](https://nimfadora.github.io/pr/com.puppycrawl.tools.checkstyle.checks/index.html)

threshold before changes: 90%
current threshold: 93%
execution time: 03:28

### CODE REVIEW

**Comment by romani** [comment]
@Nimfadora , localization tests are failed, please fix

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheckTest.java [hunk 1]
```diff
@@ -352,6 +356,16 @@ public void testAllowEscapesForControlCharacterSetForAllCharacters() throws Exce
         verify(checkConfig, getPath("InputAllEscapedUnicodeCharacters.java"), expected);
     }
 
+    @Test
+    public void testCountMatches() throws Exception {
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheckTest.java]: please explain in javadoc comment over method why it is not possible to make a functional test.
reflection usage should be avoided as much as possible.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheckTest.java [hunk 2]
```diff
@@ -67,6 +79,25 @@ public void testNewlineLfAtEndOfFile() throws Exception {
     }
 
     @Test
+    public void testCloseRandomAccessFile() throws Exception {
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheckTest.java]: we recently decided to force code with mocks to have javadoc comment explaining a reason why it impossible to make functional test. This case is approved to me based on mock, lets just left a message to future maintainers/contributors.
please add javadoc comment on method.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheckTest.java [hunk 3]
```diff
@@ -88,6 +100,19 @@ public void testDefault() throws Exception {
         verify(checkConfig, getPath("InputUniqueProperties.properties"), expected);
     }
 
+    @Test
+    public void testCloseInputStream() throws Exception {
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheckTest.java]: please add javadoc comment to explain mock usage.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheckTest.java (hunk 1 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheckTest.java]: fixed

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheckTest.java (hunk 2 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheckTest.java]: fixed

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheckTest.java (hunk 3 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheckTest.java]: fixed

**Pull request merged**